### PR TITLE
refactor!: Rework code for refactored MessageBus Configuration

### PR DESCRIPTION
**Pull request by lenny-intel**
BREAKING CHANGE: MessageQueue renamed to MessageBus and fields changed. See v3 Migration guide.

This PR is dependent on https://github.com/edgexfoundry/go-mod-bootstrap/pull/410 

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [x] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  TBD

## Testing Instructions
Run non-secure Edgex Stack with device virtual and stop all core/support services.
Build core/support services locally
Run core/support services locally with DEBUG log level
Verify core/support services bootstrap properly and connect to the MessageBus
Verify Core Data is receiving events.

### CODE REVIEW

File: internal/core/command/main.go [hunk 1]
```diff
@@ -84,37 +84,33 @@ func Main(ctx context.Context, cancel context.CancelFunc, router *mux.Router) {
...
-		}
-		if err := messaging.SubscribeCommandResponses(ctx, router, dic); err != nil {
-			lc.Errorf("Failed to subscribe commands response from internal message bus, %v", err)
-			return false
-		}
-		if err := messaging.SubscribeCommandQueryRequests(ctx, dic); err != nil {
-			lc.Errorf("Failed to subscribe command query request from internal message bus, %v", err)
+	if configuration.MessageQueue.External.Enabled {
+		if !handlers.NewExternalMQTT(messaging.OnConnectHandler(router, dic)).BootstrapHandler(ctx, wg, startupTimer, dic) {
 			return false
```
Comment (jim-wang-intel) [inline internal/core/command/main.go:95]: do an error logging here

Comment (lenny-intel) [reply internal/core/command/main.go:95]: logging is done in the handler

**Review by jim-wang-intel — DISMISSED** [review]

File: cmd/support-notifications/res/configuration.toml [hunk 2]
```diff
@@ -73,14 +68,16 @@ Type = "consul"
...
+[MessageBus]
 Protocol = "redis"
 Host = "localhost"
 Port = 6379
 Type = "redis"
 AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
 SecretName = "redisdb"
-  [MessageQueue.Optional]
+  [MessageBus.Topics]
+  # No topic yet. Metrics topic is handler in [Writable.Telemetry] above
```
Comment (jim-wang-intel) [inline cmd/support-notifications/res/configuration.toml:79]:
```suggestion
  # No topic yet. Metrics topic is handled in [Writable.Telemetry] above
```

File: cmd/support-scheduler/res/configuration.toml [hunk 3]
```diff
@@ -73,14 +73,16 @@ Type = "consul"
...
+[MessageBus]
 Protocol = "redis"
 Host = "localhost"
 Port = 6379
 Type = "redis"
 AuthMode = "usernamepassword"  # required for redis messagebus (secure or insecure).
 SecretName = "redisdb"
-  [MessageQueue.Optional]
+  [MessageBus.Topics]
+  # No topic yet. Metrics topic is handler in [Writable.Telemetry] above
```
Comment (jim-wang-intel) [inline cmd/support-scheduler/res/configuration.toml:84]:
```suggestion
  # No topic yet. Metrics topic is handled in [Writable.Telemetry] above
```

**Review by ejlee3 — DISMISSED** [review]
LGTM!

File: cmd/support-scheduler/res/configuration.toml (hunk 3 above)
Comment (lenny-intel) [reply cmd/support-scheduler/res/configuration.toml]: fixed

File: cmd/support-notifications/res/configuration.toml (hunk 2 above)
Comment (lenny-intel) [reply cmd/support-notifications/res/configuration.toml]: fixed

**Review by jim-wang-intel — APPROVED** [review]

**Review by bnevis-i — APPROVED** [review]

**Pull request merged**